statsmodels/tsa/statespace/kalman_filter.py

```diff
@@ -712,6 +712,313 @@ def loglikeobs(self, loglikelihood_burn=None, **kwargs):
+        time_invariant = self.time_invariant
+
+        # Holding variables for the simulations
+        simulated_obs = np.zeros((nsimulations, self.k_endog),
+                                 dtype=self.dtype)
+        simulated_states = np.zeros((nsimulations+1, self.k_states),
+                                    dtype=self.dtype)
+        simulated_states[0] = initial_state
+
+        # Perform iterations to create the new time series
+        obs_intercept_t = 0
+        design_t = 0
+        state_intercept_t = 0
+        transition_t = 0
+        selection_t = 0
+        for t in range(nsimulations):
+            # Get the current shocks (this accomodates time-varying matrices)
+            if measurement_shocks is None:
+                measurement_shock = np.random.multivariate_normal(
+                    mean=np.zeros(self.k_endog), cov=self['obs_cov', :, :, t])
+            else:
+                measurement_shock = measurement_shocks[t]
+
+            if state_shocks is None:
+                state_shock = np.random.multivariate_normal(
+                    mean=np.zeros(self.k_posdef),
+                    cov=self['state_cov', :, :, t])
+            else:
+                state_shock = state_shocks[t]
+
+            # Get current-iteration matrices
+            if not time_invariant:
+                obs_intercept_t = 0 if self.obs_intercept.shape[-1] == 1 else t
+                design_t = 0 if self.design.shape[-1] == 1 else t
+                state_intercept_t = (
+                    0 if self.state_intercept.shape[-1] == 1 else t)
+                transition_t = 0 if self.transition.shape[-1] == 1 else t
+                selection_t = 0 if self.selection.shape[-1] == 1 else t
+
+            obs_intercept = self['obs_intercept', :, obs_intercept_t]
+            design = self['design', :, :, design_t]
+            state_intercept = self['state_intercept', :, state_intercept_t]
+            transition = self['transition', :, :, transition_t]
+            selection = self['selection', :, :, selection_t]
+
+            # Iterate the measurement equation
+            simulated_obs[t] = (
+                obs_intercept + np.dot(design, simulated_states[t])
+                + measurement_shock)
+
+            # Iterate the state equation
+            simulated_states[t+1] = (
+                state_intercept + np.dot(transition, simulated_states[t]) +
+                np.dot(selection, state_shock))
+
+        return simulated_obs, simulated_states[:-1]
+
+    def impulse_responses(self, steps=1, impulse=0, orthogonalized=False,
+                          cumulative=False, **kwargs):
+        """
+        Impulse response function
+
+        Parameters
+        ----------
+        steps : int, optional
+            The number of steps for which impulse responses are calculated.
+            Default is 1. Note that the initial impulse is not counted as a
+            step, so if `steps=1`, the output will have 2 entries.
```

Comment:
Good point about the default, I'll make it 10 as in VAR.

After that change, they will be roughly the same. It's a little different since these are more generic / need to fit more possible models. I'll respond to this more fully in the general PR comments.